week08/movie-app/src/components/Header.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+
+const HeaderStyle = styled.div`
+    @font-face {
+        font-family: 'HSSanTokki20-Regular';
+        src: url('https://fastly.jsdelivr.net/gh/projectnoonnu/2405@1.0/HSSanTokki20-Regular.woff2') format('woff2');
+        font-weight: normal;
+        font-style: normal;
+    }
+    font-family: 'HSSanTokki20-Regular';
+    font-size: 40px;
+    color: #00ff84;
+    display: flex;
+    align-items: center;
+    background-color: #000000;
+    height: 80px;
+    padding-left: 40px;
+`;
+
+const HowtoSort = styled.div`
+    display: flex;
+    margin-left: auto;
+    margin-right: 30px;
+    padding-right: 10px;
+
+    button {
+        margin-left: 20px;
+        background: none;
+        border: none;
+        border-radius: 7px;
+        padding: 10px;
+        color: #00ff84;
+        font-family: 'HSSanTokki20-Regular';
+        font-size: 18px;
+        cursor: pointer;
+
+        &:hover {
+            color: #000000;
+            background-color: #00ff84;
+        }
+    }
+`;
+
+const GenreList = styled.div`
+    position: absolute;
+    top: 80px;
+    right: 30px;
+    background-color: none;
+    color: #00ff84;
+    border: 1px solid #00ff84;
+    border-radius: 7px;
+    padding: 10px;
+    display: flex;
+    flex-direction: column;
+
+    button {
+        background: none;
+        border: none;
+        color: #00ff84;
+        cursor: pointer;
+        margin: 5px 0;
+    }
+`;
+
+const Header = ({ selectGenre, changeTitle }) => {
+    const [showGenres, setShowGenres] = useState(false);
+
+    const sortingFunc = (criteria) => () => {
+        console.log(criteria);
+        changeTitle(criteria);
+        setShowGenres(false);
+    };
+
+    return (
+        <HeaderStyle>
+            <div>Movillie</div>
+            <HowtoSort>
+                <button onClick={sortingFunc("")}>기본 정렬</button>
+                <button onClick={sortingFunc("rating")}>평점 순</button>
+                <button onClick={sortingFunc("year")}>개봉 순</button>
+                <button onClick={() => setShowGenres(!showGenres)}>장르 별</button>
+                {showGenres && (
+                    <GenreList>
+                        <button onClick={() => selectGenre("Drama")}>드라마</button>
+                        <button onClick={() => selectGenre("Horror")}>호러</button>
+                        <button onClick={() => selectGenre("Comedy")}>코미디</button>
+                        <button onClick={() => selectGenre("Romance")}>로맨스</button>

```

Comment:
P5: 장르 리스트 속에 장르명이 반복되고 있으니, 컴포넌트를 하나 만들고 map 으로 뿌려주면 좋을 것 같아요!

```
const genres = [
  { label: "드라마", genre: "Drama" },
  { label: "호러", genre: "Horror" },
  { label: "코미디", genre: "Comedy" },
  { label: "로맨스", genre: "Romance" },
];
```
이런 식으로 배열을 만들고, 


```suggestion
        {genres.map((item) => (
          <button key={item.genre} onClick={() => selectGenre(item.genre)}>
            {item.label}
          </button>
        ))}
```

이렇게 map을 사용하면 코드 유지보수가 더 편리할 것 같아요~!